src/Control/Distributed/Spark/RDD.hs

```diff
@@ -119,6 +158,27 @@ aggregate seqOp combOp zero rdd = do
   res :: JObject <- call rdd "aggregate" [coerce jzero, coerce jseqOp, coerce jcombOp]
   reify (unsafeCast res)
 
+-- | A version of aggregate implemented in terms of 'mapPartitions'.
+aggregate'
```

Comment:
The main motivation to bind mapPartitions is that it allows processing the elements of a partition without copying the intermediate results back and forth between Haskell and Java for each element.

If the user does not use types with a cheap marshaling from Haskell to Java, the copying cost would dominate the computation using the `aggregate` binding.

In our case we had a Data.Map.Map which was updated once per element. Copying the vector was a rather expensive operation. Another alternative could have been to use a vector-like datastructure in the Java heap and update it from the Haskell side. Another example of intermediate datastructure we have is a list of trees, which would be more laborious to reimplement in java.

But using java datastructures would also bring the question of why shouldn't we implement the whole thing in java and call it as a monolithic computation, instead of micromanaging the intermediate structure updates from Haskell, which would render useless an `aggregate` binding.